src/table.cpp

```diff
@@ -97,3 +98,38 @@ std::vector<JoinResult> Table::hashJoin(size_t index, Table &other, size_t other
 
     return result;
 }
+
+std::vector<JoinResult> Table::hashJoinRows(size_t index, const std::vector<std::size_t> &rows, Table &other, size_t other_index) {
+    std::vector<JoinResult> result;
+
+    auto &column = this->column(index)->data();
+    auto &other_column = other.column(other_index)->data();
+
+    // We weren't using multimaps before, so actually the hash join operation
+    // wasn't implemented correctly. Definitely need them for Task 3, because we have multiple
+    // rows with the same values
+    using map_t = std::unordered_multimap<uint32_t, size_t>;
+
+    // TODO: Also omitting allocating to bind the map onto a NUMA node -- was getting allocation failures
+    // Now it's probably sliced across all nodes, which is not what we want
+    map_t col_map;
```

Comment:
This should be fixed, don't know why the allocator is not working.

One workaround could be to use an `unordered_map<uint32, std::vector<size_t, allocator_t>, ..., allocator_t>`